packages/core/src/tasks/index.ts

```diff
@@ -1065,31 +1078,60 @@ export const proposeChugSplashDeployment = async (
       spinner
     )
 
-    const deploymentId = computeDeploymentId(
-      bundles.actionBundle.root,
-      bundles.targetBundle.root,
-      bundles.actionBundle.actions.length,
-      bundles.targetBundle.targets.length,
-      getDeployContractActions(bundles.actionBundle).length,
-      configUri
-    )
-
     // Verify that the deployment has been committed to IPFS with the correct bundle hash.
     await verifyDeployment(provider, configUri, deploymentId, ipfsUrl)
   }
+
   // Propose the deployment.
-  await (
-    await ChugSplashManager.propose(
-      bundles.actionBundle.root,
-      bundles.targetBundle.root,
-      bundles.actionBundle.actions.length,
-      bundles.targetBundle.targets.length,
-      getDeployContractActions(bundles.actionBundle).length,
-      configUri,
-      remoteExecution,
-      await getGasPriceOverrides(provider)
+  if (
+    process.env.CHUGSPLASH_API_KEY &&
+    ((await isLiveNetwork(provider)) ||
+      process.env.LOCAL_TEST_METATX_PROPOSE === 'true')
+  ) {
+    if (!process.env.PRIVATE_KEY) {
+      throw new Error(
+        'Must provide a PRIVATE_KEY environment variable to sign gasless proposal transactions'
+      )
+    }
```

Comment:
nit: I don't think this if-statement can ever be triggered

Reply:
To my knowledge, neither Hardhat nor Foundry require the use of the PRIVATE_KEY environment variable. If the user chose to use any other environment variable name for their private key (or used something else like a mnemonic phrase in Foundry), then our meta transaction logic would break as it relies on reading a private key from specifically this environment variable.

Reply:
Sorry I misread the original code. Disregard this